homeassistant/components/sensibo/config_flow.py

```diff
@@ -38,32 +36,67 @@ async def async_validate_api(hass: HomeAssistant, api_key: str) -> bool:
 
     try:
         async with async_timeout.timeout(TIMEOUT):
-            if await client.async_get_devices():
-                return True
+            await client.async_get_devices()
+    except AuthenticationError as err:
+        LOGGER.error("Failed to authenticate to Sensibo servers %s", err)
+        return "invalid_auth"
     except (
         aiohttp.ClientConnectionError,
         asyncio.TimeoutError,
-        AuthenticationError,
         SensiboError,
     ) as err:
-        _LOGGER.error("Failed to get devices from Sensibo servers %s", err)
-    return False
+        LOGGER.error("Failed to get devices from Sensibo servers %s", err)
+        return "cannot_connect"
+    return "validated"
 
 
-class SensiboConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+class SensiboConfigFlow(ConfigFlow, domain=DOMAIN):
     """Handle a config flow for Sensibo integration."""
 
     VERSION = 1
 
-    async def async_step_import(self, config: dict) -> FlowResult:
+    async def async_step_import(self, config: dict[str, Any]) -> FlowResult:
         """Import a configuration from config.yaml."""
 
-        self.context.update(
-            {"title_placeholders": {"Sensibo": f"YAML import {DOMAIN}"}}
-        )
         return await self.async_step_user(user_input=config)
 
-    async def async_step_user(self, user_input=None) -> FlowResult:
+    async def async_step_reauth(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle re-authentication with Sensibo."""
+        errors: dict[str, str] = {}
+
+        entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+
+        if user_input is not None:
+            api_key = user_input[CONF_API_KEY]
+            validate = await async_validate_api(self.hass, api_key)
+            if validate != "validated":
+                errors["base"] = validate
+            else:
+                existing_entry = await self.async_set_unique_id(
+                    self.context["unique_id"]
+                )
+                if existing_entry and entry:
```

Comment:
We should abort if they change accounts

Reply:
This is a bit tricky as if the api_key becomes invalid because the user removes it and comes here to enter a new api_key I would not know if it's for same account or for a different account. I can only assume it's for same account.
As you see in the data schema there is only the api key and no username or such.

Reply:
Unrelated to this PR, but I noticed when looking for a better unique id that the api returns the mac address of the unit so it would be nice to add it under `connections` in a future PR: https://developers.home-assistant.io/docs/device_registry_index?_highlight=hw_ve#device-properties

Reply:
``fetch("https://home.sensibo.com/api/v1/users/me?", {
  "headers": {
    "accept": "*/*",
    "accept-language": "en-US,en;q=0.9",
    "sec-fetch-dest": "empty",
    "sec-fetch-mode": "cors",
    "sec-fetch-site": "same-origin",
    "sec-gpc": "1"
  },
  "referrer": "https://home.sensibo.com/",
  "referrerPolicy": "strict-origin-when-cross-origin",
  "body": null,
  "method": "GET",
  "mode": "cors",
  "credentials": "include"
});``

Assuming the username isn't changeable...

You could call the `/api/v1/users/me` endpoint to get the `username` for the api token.  You'd have to add it the config entry data as well if its missing so when we do need to do reauth we have something to compare against.

Reply:
Good call. I will make two PR's for this purpose. One to bump the package to support the new endpoint to get that info and another one to migrate the config entry to use username instead.